snappy/overlord.go

```diff
@@ -0,0 +1,89 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snappy
+
+import (
+	"path/filepath"
+
+	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/progress"
+)
+
+// Overlord is responsible for the overall system state.
+type Overlord struct {
+}
+
+// Install installs the given snap file to the system.
+//
+// It returns the local snap file or an error
+func (o *Overlord) Install(snapFilePath string, origin string, inter progress.Meter, flags InstallFlags) (*SnapPart, error) {
+	return nil, ErrNotImplemented
+}
+
+// Uninstall removes the given local snap from the system.
+//
+// It returns an error on failure
+func (o *Overlord) Uninstall(sp *SnapPart, pb progress.Meter) error {
+	return ErrNotImplemented
+}
+
+// SetActive sets the active state of the given snap
+//
+// It returns an error on failure
+func (o *Overlord) SetActive(sp *SnapPart, active bool, pb progress.Meter) error {
```

Comment:
s/pb/meter/?  Does it really need a meter, though?  What's the line between operations that need a meter from those that do not?  I'm think we should really have a callback setting for creating new meters, so we don't need to decide upfront in the API whether a given function needs a meter or not.